### DISCUSSION TITLE
Use different metric than max combo for combo portion of score

### DISCUSSION THREAD

**Discussion by WitherFlower** [ppy/osu#13160] [Ideas]
**While the new scoring system is good for comparing full combo scores, it doesn't work well for comparing scores with misses.**

Here is an example : 

![image](https://user-images.githubusercontent.com/52672543/95011439-00b5a300-0631-11eb-8f06-b80defc16bd1.png)

First score is a one miss play, with a miss in the middle of the map
Second score has many misses in the first half (one every 30 combo or so), and no misses in the second half

Here is a visual representation of combo over time for both plays : 

![image](https://user-images.githubusercontent.com/52672543/95011661-7d954c80-0632-11eb-867e-5f899e9c6ed0.png)

However, the two scores values are almost the same, and this is a problem, because I believe this is unfair for the player that has only one miss in the map.

This comes from the fact that the two scores have the same maximum combo, and that is the **only** parameter taken in account for combo portion calculation.

### I'm proposing three solutions for this problem : 

- The first solution would be to calculate the combo portion based on the **average** combo in the map, however it would be hard to convert scores with no replay. Not impossible, because we could figure out the math to approximate the average combo with scorev1, but that's hard to do.

 - The second solution would be to make misses count as negative values (like in scorev2), however I believe that it's not enjoyable to see score decreasing from a player's standpoint

 - The final solution, which is probably the simplest, would be to make misses count in the calculation of combo portion, with a decreasing exponential factor for example. The "problem" of score decreasing when you miss could be avoided by doing something like this : 
`missFactor = exp( -(remainingCombo + missCount) / scaling)`
with scaling determining how severe the miss punishment is.

I hope this issue will be considered, so we can have a flawless scoring system !

**Comment by peppy** [comment]
I'm not sure this is an issue. The information you've given is not really helpful in understanding the issue you are bringing up. Can you show the actual score screens rather than some limited detail? Or include the replays? Or anything really.

**Comment by WitherFlower** [comment]
Here are the score screens : 
![image](https://user-images.githubusercontent.com/52672543/95012205-dd8df200-0636-11eb-9eb1-2b13319ed98e.png)

**Comment by peppy** [comment]
I think this looks fine? You're comparing two average scores which are 33-35% of possible score. They are both average scores in their own ways (one has lower accuracy, while the other has more misses).

**Comment by enoslayd** [comment]
i thought the combo on scores represent the highest combo that a player achieve in a play?

**Comment by peppy** [comment]
That is correct thinking.

**Comment by enoslayd** [comment]
then, what is the problem?

**Comment by WitherFlower** [comment]
> one has lower accuracy, while the other has more misses
The two scores have almost the same accuracy (15x100 vs 12x100 and 1x50), but the 8 misses only make for a tiny difference in  score, as they are only represented through accuracy.

The problem is that score #1 is almost a full combo, while score #2 is nowhere near a full combo, however they are pretty much identical, and I think that this is not very fair

-----

Imagine a map that has an easy first part, and a harder second part.
Player A hits almost everything, except a single note in the beginning of the harder part
Player B hits the first part prefectly, but doesn't play good at all on the second part and has many misses

The score difference would only come from accuracy, and so it wouldn't be really significant despite the two plays having a clear difference in skill, and I think this is not fair

**Comment by peppy** [comment]
The original poster is saying that "more combo breaks should result in a lower score", basically implying that misses aren't visible enough compared to more "meh" / "ok" results. It's a very nuanced point of view which is more about the relative scoring of results rather than combo.

**Comment by peppy** [comment]
Do you have standardised scoring turned off? Those scores both seem much lower than they should be.

**Comment by WitherFlower** [comment]
These scores were made with standardised scoring on the latest release. The scores are lower because I used NoFail.

**Comment by MandL27** [comment]
Throwing my 2c into the ring, since I've been thinking about this a bit of late: what about tracking second longest, third longest, etc. combos, then adding them all up with weighting?

Taking OP's two examples, and assuming max combo on the map is 586, example 1 would be a 319 combo at full weight and then a 266 combo at some lower weight, and example 2 would be a 320 combo at full weight, then several 30ish combos at increasingly lower weights.

Further assuming that weightings are a linear 100-80-60-40-20 for the five longest combos, example 1 would get a combo portion of (319 + (266 \* 0.8)) / 586 * 700000 = 635255, and example 2 would get a combo portion of (320 + (30 \* 0.8) + (30 * 0.6) + ...) / 586 * 700000 = 452730. The numbers could probably be adjusted, but the concept here feels to me like a workable compromise between max combo and miss count (the two extrema I think are at play here).

edit: Reread the OP and realized this is similar to the first idea proposed there, though I'm specifically referring to only counting the longest few (with the weighting used here, five) combos and discarding the rest.

**Comment by bdach** [comment]
To cite part of this discussion that took place on discord, we'd be looking to make the scoring easily understandable. Doing calculations like that would run contrary to that.

**Comment by MandL27** [comment]
That's a good point. A simpler approach could be averaging miss count and max combo, since both of those are shown on the result screen.
